src/data/dfinsupp/well_founded.lean

```diff
@@ -0,0 +1,190 @@
+/-
+Copyright (c) 2022 Junyan Xu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Junyan Xu
+-/
+import data.dfinsupp.lex
+import order.game_add
+import order.antisymmetrization
+import set_theory.ordinal.basic
+
+/-!
+# Well-foundedness of the lexicographic and product orders on `dfinsupp` and `pi`
+
```

Comment:
Can you expand on this by adding a more precise description of the main results (including their names)?

Reply:
Done!